### ISSUE TITLE
systemd-254.7: systemd-userdbd.service fails to start since upgrading to kernel 6.6

### ISSUE THREAD

**Issue by pacho2** [systemd/systemd#30535]
Hello,

I recently updated a machine from kernel 6.4.3 to 6.6.7 in Gentoo and I noticed that systemd-userdbd.service started to fail to start at boot time (https://bugs.gentoo.org/920331). When I reviewed service status, it seems to point to a permission error due to, maybe, it trying to start too soon (otherwise, service starts fine once system is booted)
```
# systemctl status systemd-userdbd.service 
× systemd-userdbd.service - User Database Manager
     Loaded: loaded (/usr/lib/systemd/system/systemd-userdbd.service; indirect; preset: disabled)
     Active: failed (Result: exit-code) since Tue 2023-12-19 18:45:05 CET; 2min 33s ago
TriggeredBy: × systemd-userdbd.socket
       Docs: man:systemd-userdbd.service(8)
    Process: 278 ExecStart=/usr/lib/systemd/systemd-userdbd (code=exited, status=226/NAMESPACE)
   Main PID: 278 (code=exited, status=226/NAMESPACE)
        CPU: 410us

dic 19 18:45:05 hp-isidro systemd[1]: Starting systemd-userdbd.service...
dic 19 18:45:05 hp-isidro (-userdbd)[278]: systemd-userdbd.service: Failed to set up mount namespacing: /run/systemd/mount-rootfs/dev: Read-only file system
dic 19 18:45:05 hp-isidro (-userdbd)[278]: systemd-userdbd.service: Failed at step NAMESPACE spawning /usr/lib/systemd/systemd-userdbd: Read-only file system
dic 19 18:45:05 hp-isidro systemd[1]: systemd-userdbd.service: Main process exited, code=exited, status=226/NAMESPACE
dic 19 18:45:05 hp-isidro systemd[1]: systemd-userdbd.service: Failed with result 'exit-code'.
dic 19 18:45:05 hp-isidro systemd[1]: Failed to start systemd-userdbd.service.
dic 19 18:45:05 hp-isidro systemd[1]: systemd-userdbd.service: Start request repeated too quickly.
dic 19 18:45:05 hp-isidro systemd[1]: systemd-userdbd.service: Failed with result 'exit-code'.
dic 19 18:45:05 hp-isidro systemd[1]: Failed to start systemd-userdbd.service.
```

I reviewed again the new options since kernel 6.4 to 6.6 and I couldn't find anything that could be related with the problem or a stronger "hardening" being enabled now :/

When searching, I found this similar issue:
https://github.com/coreos/fedora-coreos-tracker/issues/1296
https://forums.gentoo.org/viewtopic-p-8807305.html

But I still fail to understand what changed with kernel 6.6, as 6.4 is still working fine on the same computer.

Thanks for your help

**Comment by mrc0mmand** [comment]
I think this is the same issue as https://github.com/systemd/systemd/issues/30372 that was fixed in https://github.com/systemd/systemd/pull/30375 and should get to v254.8 once https://github.com/systemd/systemd-stable/pull/351 lands.

**Comment by pacho2** [comment]
Thanks for the info

Sadly, I have tried to apply https://patch-diff.githubusercontent.com/raw/systemd/systemd-stable/pull/351.patch over 254.7 (skipping the first CentOS related patch as otherwise it wasn't applying) and it fails in the same way :(

**Comment by mrc0mmand** [comment]
Ah, in that case this might be https://github.com/systemd/systemd/pull/29043 which was fixed by https://github.com/systemd/systemd/pull/29343 (and this change is not lined up for v254.8) - could you give that one a try?

**Comment by pacho2** [comment]
Yeah, the last one fixes the problem :D

Thanks a lot!

**Comment by pacho2** [comment]
Will it be included in 254.8 or we will need to apply it downstream until 254.9? Thanks for the info

**Comment by mrc0mmand** [comment]
> Will it be included in 254.8 or we will need to apply it downstream until 254.9? Thanks for the info

Nope, this won't be a part of 254.8.
